src/backups.py

```diff
@@ -369,6 +376,10 @@ def _on_s3_credential_changed(self, event: CredentialsChangedEvent):
             logger.debug("Cannot set pgBackRest configurations, missing configurations.")
             return
 
+        # Verify the s3 relation only on the leader
+        if not self.charm.unit.is_leader():
+            return
+
```

Comment:
Do the init checks only on the leader unit. Followers should be updated from the peer change events.